corehq/apps/hqwebapp/views.py

```diff
@@ -211,6 +212,32 @@ def password_change(req):
     return render(req, template_name, {"form": password_form})
 
 
+# Corresponds to django.contrib.auth.views.password_reset_confirm()
+# Handles when a user clicks the link in their "reset password" e-mail and prompts for a new password
+def domain_reset_pwd(request, domain, uidb64, token, template_name='login_and_password/reset_pwd.html'):
+    domain_instance = Domain.get_by_name(domain)
+    if not domain_instance:
+        raise Http404
+    request.project = domain_instance
+
+    extra_context = {
+        'domain': domain,
+        'display_name': domain_instance.display_name() or domain,
+        'current_page': {'page_name': _('Welcome to %s!') % domain_instance.display_name()},
```

Comment:
might be nicer for the translators if this was named, otherwise there is no context